src/gpl/src/mbff.h

```diff
@@ -0,0 +1,111 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (c) 2023, Precision Innovations Inc.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+///////////////////////////////////////////////////////////////////////////////
+
+#include <vector>
+
+#pragma once
+
+struct Point {
+    float x = 0, y = 0;
+};
+
+struct Tray {
+    Point pt;
+    std::vector<Point> slots;
+    int cand[70];
+};
+
+struct Flop {
+    Point pt;
+    int idx = 0;
+    float prob = 0;
+
+    bool operator<(Flop &a) { return prob < a.prob; }
+};
+
+struct Path {
+    int a = 0, b = 0;
+};
+
+namespace gpl {
+
+class MBFF {
+
+  public:
+    MBFF(const int num_flops, const int num_paths, const std::vector<float>& x,
+         const std::vector<float>& y, const std::vector<std::pair<int, int> >& paths,
+         const int threads);
+    ~MBFF();
+
+    void Run(const int mx_sz, const float alpha, const float beta);
+
+  private:
+
+    float GetDist(const Point& a, const Point& b);
```

Comment:
If you use odb::dbPoint this method would already be defined.

However, assuming we needed a point class it would make more sense to define Point as a class with a distance method. 

The object oriented programming model attempts to merge code with data. So in general it's best to keep methods as close to their data container as possible